MushroomObserver/mushroom-observer
# Add user option for persistent content filters.

Gives user the option - via Preferences - to always exclude imageless or unvouchered Observations from Activity Feed and search results. 
(Can override via search bar filters or Advanced Search checkboxes.)

Basic Manual Test Script
* Create 4 Observations, as follows
  (a) one with an image and a specimen;
  (b) one with an image but without specimen;
  (c) one with a specimen but without images;
  (d) one without images or specimen.
* Refresh Activity Feed.

Result: all 4 Observations should be shown.

- Turn on has_images persistent filter: in /account/prefs, check the box next to “Hide Observations with no images.”, then Save Edits
- Refresh Activity Feed.

Result: Of those 4 Observations, only (a) and (b) should appear.  “(filtered)” appears next to page title. 

- Turn off has_images persistent filter, turn on has_specimen filter, Save Edits.
- Refresh Activity Feed.  

Result: Of those 4 Observations, only (a) and (c) should appear. (filtered)” appears next to page title. 

- Turn on both filters, Save Edits.
- Refresh Activity Feed.  

Result: Of those 4 Observations, only (d) should appear. (filtered)” appears next to page title. 

Note: this is a simple script.  These filters persist site-wide for all Observation queries.  For instance, with any of these filters on, from any /name/show_name/nnnnnn page, click More Observations.  The results are filtered (and that is shown in the page title).
